lib/services/role.go

```diff
@@ -348,6 +348,18 @@ func ApplyTraits(r types.Role, traits map[string][]string) types.Role {
 		r.SetHostGroups(condition,
 			applyValueTraitsSlice(r.GetHostGroups(condition), traits, "host_groups"))
 
+		inHostSudoers := r.GetHostSudoers(condition)
+		var outHostSudoers []string
+		for _, entry := range inHostSudoers {
+			vals, err := ApplyValueTraits(entry, traits)
+			if err != nil && !trace.IsNotFound(err) {
+				log.Warnf("did not apply trait to sudoers entry: %v", err)
+				continue
```

Comment:
Are we SURE we want to continue here, instead of returning an error? Won't this explode horribly later on, as the sudoers file will have a `{{foo}}` where there shouldn't be one?